doc/design/design_model_zoo.md

```diff
@@ -0,0 +1,187 @@
+# Model Zoo
+
+SQLFlow model zoo is a place to store model definitions, pre-trained model weights and model documentations. You can directly train, predict, analyze using one of the models using SQLFlow, or you can do model fine-tune, transfer learning to use the model to fit your dataset.
+
+SQLFlow should support below features to support common cases in machine learning:
+
+1. Host model definition and pre-trained weights in `sqlflow.org`. e.g. `sqlflow.org/modelzoo/iris_dnn_128x32` points to a directory containing a model definition of `DNNClassifier` with 128, 32 hidden layers and pre-trained weights using the iris dataset.
+1. Download pre-trained model to predict a dataset:
+   ```sql
+   SELECT * FROM iris.predict_samples
+   PREDICT predict_result.class
+   USING sqlflow.org/modelzoo/iris_dnn_128x32;
+   ```
+1. Train a model from scratch using the model definition:
+   ```sql
+   SELECT * FROM iris.new_iris_train
+   TRAIN sqlflow.org/modelzoo/iris_dnn_128x32
+   INTO modeldb.my_iris_dnn_model;
+   ```
+1. Transfer learning to fit a new dataset with the same features:
+   ```sql
+   SELECT * FROM iris.transfer_iris_samples
+   TRAIN sqlflow.org/modelzoo/iris_dnn_128x32
+   USING sqlflow.org/modelzoo/iris_dnn_128x32
+   INTO modeldb.my_iris_dnn_model;
+   ```
+1. Fine-tune the pre-trained model:
+   ```sql
+   SELECT * FROM iris.train
+   TRAIN sqlflow.org/modelzoo/iris_dnn_128x32
+   WITH model.learning_rate=0.001, model.learning_rate_decay="cosine_decay" ...
+   INTO modeldb.my_iris_dnn_model_fine_tune;
+   ```
+
+## The Model Zoo Hosting Service
+
+The model zoo hosting service is a file service that can be accessed from the internet.
+It serves all available models and corresponding pre-trained weight. We may need to use
+a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network) service if the traffic
+becomes large.
+
+Each model is saved on the server under a unique directory like: `sqlflow.org/modelzoo/iris_dnn_128x32`.
+The directory name is responsible to explain the model's type, network structure and which dataset is
+used to train the pre-trained weights. You can access `sqlflow.org/modelzoo/iris_dnn_128x32/README.md`
+from the browser to get the model's full documentation. All models under `sqlflow.org/modelzoo` are
+developed under `https://github.com/sql-machine-learning/models` weights is only stored under
+`sqlflow.org` but not under Github.
+
+The content of the directory should be like:
+
+```
+iris_dnn_128x32
+   - model_meta.json  # model information useful for load and run.
+   - README.md        # model documents.
+   - requirments.txt  # python package dependency for the model.
+   - model_def.py     # python file of model definition (TensorFlow Estimator model or Keras model).
+   - model/           # the saved TensorFlow/Keras/XGBoost model.
```

Comment:
Github repo might not be a suitable place to store the actual artifacts of models since the models can get really large and Github is not suitable for large files. We need to consider a third-party storage for this.